core/src/main/java/org/apache/carbondata/core/cache/CarbonLRUCache.java

```diff
@@ -305,4 +312,15 @@ public void clear() {
       lruCacheMap.clear();
     }
   }
+
+  /**
+   * Check if LRU cache setting is bigger than max memory of jvm.
+   * if LRU cache is bigger than max memory of jvm when query for a big segments table,may cause JDBC server crash.
+   * @return true LRU cache is bigger than max memory of jvm, false otherwise
+   */
+  private boolean isBeyondMaxMemory(){
+    long mSize = Runtime.getRuntime().maxMemory();
+    long lruSize = lruCacheMemorySize * 1024 * 1024;
```

Comment:
better to use BYTE_CONVERSION_CONSTANT instead of 1024 * 1024

Reply:
ok